src/components/AddItems.jsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useState } from 'react';
+import { useStateWithStorage } from '../utils';
+import { addItem } from '../api';
+
+const soonDate = 7;
+const kindOfSoonDate = 14;
+const notSoonDate = 30;
+
+export function AddItems() {
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+	const [itemName, setItemName] = useState('');
+
+	const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+
+	const handleTextChange = (event) => {
+		return setItemName(event.target.value);
+	};
+
+	const handleChange = (event) => {
+		const numberOfDays = parseInt(event.target.value);
+		setDaysUntilNextPurchase(numberOfDays);
+	};
+
+	const handleSubmit = useCallback(
```

Comment:
Curious about the use of the `useCallback` hook here? It seems like this function is only created and invoked when the form is submitted! We could consider simplifying the code unless there's a specific reason to memoize the function!